dialogue-core/src/test/java/com/palantir/dialogue/core/DialogueChannelsTest.java

```diff
@@ -155,6 +159,34 @@ public void bad_channel_throwing_an_error_still_returns_a_future() {
         assertThatThrownBy(() -> Futures.getUnchecked(future)).hasCauseInstanceOf(NoClassDefFoundError.class);
     }
 
+    @Test
+    @SuppressWarnings("FutureReturnValueIgnored") // intentionally spawning a bunch of throwaway requests
+    void live_reloading_an_extra_uri_allows_queued_requests_to_make_progress() {
```

Comment:
like this is the sort of test that feels weird to me. we should be able to verify in the queue tests that when the delegate channel has capacity it's able to make progress, and it potentially has to force itself to try to make progress every so often, rather than verifying that each way an underlying channel may start to have new capacity allows the queued channel to make progress.